From: Thorvald Natvig <thorvald@natvig.com>
Subject: fix XEvIE

For all the replys but ProcXevieQueryVersion, the rep.length field isn't
set to zero, which makes the client misbehave when it gets the reply.

Index: xorg-server/Xext/xevie.c
===================================================================
--- xorg-server.orig/Xext/xevie.c
+++ xorg-server/Xext/xevie.c
@@ -213,6 +213,7 @@
     
     xevieModifiersOn = FALSE;
 
+    rep.length = 0;
     rep.type = X_Reply;
     rep.sequence_number = client->sequence;
     WriteToClient (client, sizeof (xXevieStartReply), (char *)&rep);
@@ -232,6 +233,7 @@
         XevieEnd(xevieClientIndex);
     }
 
+    rep.length = 0;
     rep.type = X_Reply;
     rep.sequence_number = client->sequence;
     WriteToClient (client, sizeof (xXevieEndReply), (char *)&rep);
@@ -250,6 +252,7 @@
         return BadAccess;
 
     xE = (xEvent *)&stuff->event;
+    rep.length = 0;
     rep.type = X_Reply;
     rep.sequence_number = client->sequence;
     WriteToClient (client, sizeof (xXevieSendReply), (char *)&rep);
@@ -294,6 +297,7 @@
         return BadAccess;
 
     xevieMask = (long)stuff->event_mask;
+    rep.length = 0;
     rep.type = X_Reply;
     rep.sequence_number = client->sequence;
     WriteToClient (client, sizeof (xXevieSelectInputReply), (char *)&rep);
